fixes: size of group members for 'non-admin group viewer' should be less

vedant-jain03 opened the pull request:
Fixes #2582 

#### Describe the changes you have made in this PR -
- Added a new class for 'groups-members-card-viewer' where the width: 200px and height: auto
- If the visiter is not admin the grid is now divided into 3 but it should be 4 according to width size.
- embed ruby if condition before remove-button div.

### Screenshots of the changes (If any) -
- Before
![bandicam 2021-11-03 21-15-24-473](https://user-images.githubusercontent.com/76901313/140094201-bd41d2c5-5893-4a2c-aa33-90a894d0edce.jpg)

- After
![bandicam 2021-11-03 21-15-39-876](https://user-images.githubusercontent.com/76901313/140094225-b5f85a48-9fcc-4a2f-ac74-6c6646fb65dc.jpg)


Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

vedant-jain03 commented:
@gr455 @tachyons @nitin10s Kindly Review.

nitin10s reviewed (changes requested):
Looks good, can we increase the margin between the icon and the name. Also, please slightly increase the left padding from the icon. 

The icon doesn't appear vertically centered, can we fix that?

vedant-jain03 commented:
> Looks good, can we increase the margin between the icon and the name. Also, the icon doesn't appear vertically centered, can we fix that?

Yess Sure.

vedant-jain03 commented:
![Screenshot (171)](https://user-images.githubusercontent.com/76901313/141933168-86fa623a-45d7-419d-ae65-d88071f10d05.png)

1. Added margin-right: 20px
2. Make user icon center
3. Added tooltiptext feature for showing the full name of the user in hover.

@nitin10s Kindly Review.

vedant-jain03 commented:
![Screenshot (174)](https://user-images.githubusercontent.com/76901313/141979692-3088e553-8f45-4f56-bb43-4ec7b4fd2998.png)

1. I have added a new class for non-admin-user `groups-members-card-details-non-user` as we don't want to messup with `groups-members-card-details` class and added a conditional check to implement a class for the respective user.
2. Added margin-right for user icon 15px.
3. Added tooltiptext for better visibility of username as their is `text-overflow: ellipse`.
4. center user-icon.

@nitin10s Kindly Review.

nitin10s reviewed (changes requested):
There is a lot of space in the right that we can utilize before using overflow: ellipsis.

nitin10s commented:
Everything else looks great!

vedant-jain03 commented:
> There is a lot of space in the right that we can utilize before using overflow: ellipsis.

Sure

vedant-jain03 commented:
![Screenshot (177)](https://user-images.githubusercontent.com/76901313/143586552-9fc42e29-1bef-4d4e-8426-c3e00332e06d.png)

Change `col-7` to `col-11` for better space utilization.

vedant-jain03 commented:
> There is a lot of space in the right that we can utilize before using overflow: ellipsis.

Kindly Review @nitin10s

nitin10s commented:
The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline?

vedant-jain03 commented:
> The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline? Can we please fix that?

@nitin10s I think that is because of gap between user icon and name, as you proposed in this : https://github.com/CircuitVerse/CircuitVerse/pull/2583#pullrequestreview-805636606
If you want I can decrease the gap as well ?

vedant-jain03 commented:
> The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline? Can we please fix that?

Sure we can fix that 👍🏼

vedant-jain03 commented:
![bandicam 2021-11-30 21-08-03-298](https://user-images.githubusercontent.com/76901313/144078637-9e24e9df-edab-484a-8963-85672ef7c92b.jpg)

@nitin10s Kindly review

nitin10s reviewed (approved):
@tachyons we can merge

tachyons commented on app/views/groups/_group_members.html.erb:
```diff
@@ -1,24 +1,25 @@
 <% group.users.each do |user| %>
-  <div class="col-xs-12 col-sm-12 col-md-6 col-lg-4 groups-members-card-container">
-    <div class="groups-members-card">
+  <div class="<%= policy(group).admin_access? ? 'col-xs-12 col-sm-12 col-md-6 col-lg-4 groups-members-card-container' : 'col-xs-12 col-sm-12 col-md-6 col-lg-3 groups-members-card-container' %>">
```
Consider moving this to a helper as this is being used multiple places

vedant-jain03 replied:
Sure

vedant-jain03 replied:
@tachyons Added the helpers

vedant-jain03 commented:
@tachyons Kindly review

gr455 reviewed (approved):
Looks good to me. @tachyons You can merge this after you take a final look
